src/sugar3/speech.py

```diff
@@ -382,6 +407,10 @@ def __pipe_message_cb(self, bus, message):
             self._pipeline = None
             power.get_power_manager().restore_suspend()
             self.emit('stop')
+        elif message.type is Gst.MessageType.ELEMENT and \
+                message.get_structure().get_name() == 'espeak-mark':
+            mark_value = message.get_structure().get_value('mark')
+            self.emit('mark', int(mark_value))
```

Comment:
Why is the mark value being converted to int?

Reply:
A mistake I made as the context to which I applied it to used the number of a word which it gets from the name attribute of the mark tag so this way the method being called will know which word to highlight. I'll change it and restrict that to the context in which it's being used, thanks.